reader.go

```diff
@@ -93,18 +121,53 @@ func (r *Reader) NumRead() int {
 
 // LastModified returns the last modified time of the remote feed.
 func (r *Reader) LastModified() (time.Time, error) {
-	lastMod, err := remoteLastModified(r.ctx, r.remote)
-	return lastMod.Time(), err
+	var lastMod timestamp
+	for _, remote := range r.remotes {
+		t, err := remoteLastModified(r.ctx, remote)
+		if err != nil {
+			return time.Time{}, err
+		}
+		if t > lastMod {
+			lastMod = t
+		}
+	}
+
+	return lastMod.Time(), nil
 }
 
 // Close closes the reader.
 func (r *Reader) Close() error {
-	var err error
-	if r.fd != nil {
-		if e := r.fd.Close(); e != nil {
-			err = e
-		}
+	if r.cur != nil {
+		return r.cur.Close()
+	}
+	return nil
+}
+
+type streamReader struct {
+	remote *bfs.Object
+	opt    ReaderOptions
+	ctx    context.Context
+
+	br io.ReadCloser // bfs reader
+	cr io.ReadCloser // compression reader
+}
+
+// Read reads raw bytes from the feed.
+func (r *streamReader) Read(p []byte) (int, error) {
+	if err := r.ensureOpen(); err != nil {
+		return 0, err
+	}
+	n, err := r.cr.Read(p)
+	// only return EOF once all data has already been read
+	if n > 0 && errors.Is(err, io.EOF) {
```

Comment:
I have problems understanding this check/branch.

This thing (`streamReader`) seem to operate a single reader, just doing `ensureOpen` before any reads.

The underlying `Read(...)` call should take care of all this, looks like you're trying to shift the logic here from the "parent"/"caller" `feedx.Reader.Read`.

This seems like a hack to me, better do the checks in `feedx.Reader.Read` and leave this one dumb (just return underlying `n, err`).